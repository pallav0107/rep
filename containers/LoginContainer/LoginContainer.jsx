import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { reduxForm } from 'redux-form';
import { Flex } from 'grid-styled';
import { connect } from 'react-redux';
import { compose } from 'recompose';
import { withRouter } from 'react-router-dom';
import { Login } from '../../components/Login';
import axios from 'axios';

export class LoginContainer extends Component {
  submitHandler = (data, { action }) => {
    this.authenticate(data);
  };

  authenticate(data) {
    this.props.history.push('facilite');
    const { userName, password } = data;
    // axios({
    //   url: `http://localhost:8080/exp/staff/core/v1/login?username=${userName}&password=${password}`,
    //   method: 'GET',
    // })
    //   .then(response => {
    //     if (
    //       response !== undefined &&
    //       response !== null &&
    //       response !== '' &&
    //       response.data !== undefined &&
    //       response.data !== null &&
    //       response.data !== '' &&
    //       response.data.result !== undefined &&
    //       response.data.result !== null &&
    //       response.data.result !== '' &&
    //       response.data.result.length !== 0
    //     ) {
    //       var payload = {
    //         isLoggedIn: true,
    //         username: userName,
    //         password: password,
    //       };
    //     } else {
    //       var payload = {
    //         isLoggedIn: false,
    //         username: userName,
    //         password: password,
    //       };
    //     }
    //     this.props.onLoginClick(payload);
    //   })
    //   .then(error => {
    //   console.log(error);
    //   });
  }

  componentWillReceiveProps(nextProps) {
    const { isLoggedIn } = nextProps;
    if (isLoggedIn) {
      this.props.history.push('userhome');
    }
  }

  render() {
    const { handleSubmit } = this.props;
    return (
      <form onSubmit={handleSubmit(this.submitHandler)} onKeyPress={this.onKeyPress}>
        <Flex>
          <Login handleSubmit={this.handleSubmit} />
        </Flex>
      </form>
    );
  }
}

const mapStateToProps = state => ({
  isLoggedIn: state.reducer.isLoggedIn,
  dateError: state.reducer.dateError
});

const mapDispatchToProps = dispatch => ({
  onLoginClick: data => {
    dispatch({ type: 'LOGIN', value: data })
    onLogError: dateError => dispatch({ type: 'LOG_DATE_ERROR', dateError })
  },
});

LoginContainer.propTypes = {
  handleSubmit: PropTypes.func,
  onSubmit: PropTypes.func,
  onLogError:PropTypes.func
};

LoginContainer.defaultProps = {
  handleSubmit: () => { },
  onSubmit: () => { },
  onLogError: () => { },
};

const LoginContainer1 = compose(
  withRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  reduxForm({
    form: 'Login',
    destroyOnUnmount: false,
  })
)(LoginContainer);

export default LoginContainer1;
