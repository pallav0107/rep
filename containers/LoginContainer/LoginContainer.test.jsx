import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import LoginContainer from './LoginContainer';
import configureMockStore from 'redux-mock-store';
import "../../setupTests"
import 'raf/polyfill';

const mockStore = configureMockStore();

describe('IdentityContainer', () => {
  it('should render', () => {
    const store = mockStore({
      config: {
        appName: 'test',
      },
    });
    const component = (
      <Provider store={store}>
        <LoginContainer />
      </Provider>
    );
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
  });
});
