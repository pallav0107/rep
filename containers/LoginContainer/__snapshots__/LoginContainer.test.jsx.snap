// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdentityContainer should render 1`] = `<withRouter(Connect(ReduxForm)) />`;
